src/module/apps/compendium-browser/index.ts

```diff
@@ -737,33 +739,41 @@ class CompendiumBrowser extends Application {
                 const maxValue = currentTab.totalItemCount ?? 0;
                 if (currentValue < maxValue) {
                     currentTab.scrollLimit = Math.clamped(currentValue + 100, 100, maxValue);
-                    this.renderResultList(html, list, currentValue);
+                    this.renderResultList({ list, start: currentValue });
                 }
             }
         });
 
         // Initial result list render
-        this.renderResultList(html, list);
+        this.renderResultList({ list });
     }
 
     /**
      * Append new results to the result list
-     * @param html The Compendium Browser app HTML
-     * @param list The result list HTML element
-     * @param start The index position to start from
+     * @param options Render options
+     * @param options.list The result list HTML element
+     * @param options.start The index position to start from
+     * @param options.replace Replace the current list with the new results?
      */
-    private async renderResultList(html: HTMLElement, list: HTMLUListElement, start = 0): Promise<void> {
+    private async renderResultList({ list, start = 0, replace = false }: RenderResultListOptions): Promise<void> {
         const currentTab = this.activeTab !== "settings" ? this.tabs[this.activeTab] : null;
+        const html = this.element[0];
         if (!currentTab) return;
 
+        if (!list) {
+            const listElement = html.querySelector<HTMLUListElement>(".tab.active ul.item-list");
+            if (!listElement) return;
+            list = listElement;
+        }
+
         // Get new results from index
         const newResults = await currentTab.renderResults(start);
         // Add listeners to new results only
         this.activateResultListeners(newResults);
         // Add the results to the DOM
         const fragment = document.createDocumentFragment();
         fragment.append(...newResults);
-        list.append(fragment);
+        replace ? list.replaceChildren(fragment) : list.append(fragment);
```

Comment:
Dunno how I feel about ternary's for evaluation without resolving to a value, and especially for one with side effects.